### PR TITLE
Fixes #17816 - adds rubocop support

### PR DESCRIPTION
**Pull request by adamruzicka**
based on @ohadlevy 's #219

### CODE REVIEW

**Comment by adamruzicka** [comment]
[test]

File: app/models/foreman_tasks/task/dynflow_task.rb [hunk 1]
```diff
@@ -104,28 +94,27 @@ def get_humanized(method)
...
-               :state => data[:state].to_s,
-               :result => data[:result].to_s)
+      new(:external_id => execution_plan_id,
+          :state => data[:state].to_s,
+          :result => data[:result].to_s)
     end
 
-    private
-
+    private_class_method :model_name
```
Comment (iNecas) [inline app/models/foreman_tasks/task/dynflow_task.rb]: I'm not sure we want this method to actually be private? Based on the history, seems this should be used by external code (routes generation)

Comment (iNecas) [inline app/models/foreman_tasks/task/dynflow_task.rb]: I don't see this methods being private in foreman

File: lib/foreman_tasks/tasks/export_tasks.rake [hunk 2]
```diff
@@ -51,13 +50,11 @@ DESC
...
       def template(filename)
-        File.join(Gem::Specification.find_by_name("dynflow").gem_dir, 'web', 'views', "#{filename}.erb")
+        File.join(Gem::Specification.find_by(name: 'dynflow').gem_dir, 'web', 'views', "#{filename}.erb")
       end
 
       def erb(file, options = {})
-        unless @cache[file]
-          @cache[file] = Tilt.new(template(file))
-        end
+        @cache[file] = Tilt.new(template(file)) unless @cache[file]
```
Comment (iNecas) [inline lib/foreman_tasks/tasks/export_tasks.rake]: Does rubocop have anything againts `@cache[file] ||= Tilt.new(template(file))`?

File: app/models/foreman_tasks/task/dynflow_task.rb (hunk 1 above)
Comment (adamruzicka) [inline app/models/foreman_tasks/task/dynflow_task.rb]: So we can drop the `private` completely, right?

File: lib/foreman_tasks/tasks/export_tasks.rake (hunk 2 above)
Comment (adamruzicka) [inline lib/foreman_tasks/tasks/export_tasks.rake]: Apparently not

File: app/lib/actions/proxy_action.rb [hunk 3]
```diff
@@ -17,8 +17,8 @@ def plan(proxy, klass, options)
       plan_self(options.merge(:proxy_url => proxy.url, :proxy_action_name => klass.to_s))
     end
 
-    def run(event = nil)
-      with_connection_error_handling(event) do |event|
+    def run(an_event = nil)
```
Comment (adamruzicka) [inline app/lib/actions/proxy_action.rb]: Disabled  the cop enforcing this altogether

File: app/models/foreman_tasks/task/dynflow_task.rb (hunk 1 above)
Comment (iNecas) [inline app/models/foreman_tasks/task/dynflow_task.rb]: yes

**Comment by iNecas** [comment]
Waiting for jenkins.

**Comment by adamruzicka** [comment]
Jenkins is green, merging. Thanks for reviewing @iNecas !

**Pull request merged**